src/openvic-simulation/scripts/Condition.hpp

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include "openvic-simulation/types/IdentifierRegistry.hpp"
+#include <variant>
+#include <vector>
+
+namespace OpenVic {
+	struct ConditionManager;
+
+	// Condition: the definition of an individual condition or scope
+	// AND, OR and NOT are conditions with complex type
+	// Scopes are treated as ANDs
+	struct Condition : HasIdentifier {
+		friend struct ConditionManager;
+
+		enum class value_type_t {
+			IDENTIFIER,
+			STRING,
+			BOOLEAN,
+			INTEGER,
+			REAL,
+			COMPLEX
+		};
+
+		enum class scope_t {
+			GLOBAL,
+			COUNTRY,
+			PROVINCE,
+			POP
+		};
+
+	private:
+		const value_type_t PROPERTY(value_type);
+		std::set<scope_t> PROPERTY(scopes);
+
+		Condition(std::string_view new_identifier, value_type_t new_value_type, std::set<scope_t>&& new_scopes);
+
+	public:
+		Condition(Condition&&) = default;
+	};
+
+	// ConditionNode: represents a node, which is a map of individual conditions associated with a value, tied by
+	// a logical relationship. Simple conditions are considered to be single-element ANDs.
+	struct ConditionNode {
+		using string_t = std::string;
+		using boolean_t = bool;
+		using integer_t = int32_t;
+		using real_t = fixed_point_t;
+		//TODO add complex types
+		using condition_set_t = std::set<ConditionNode>;
```

Comment:
```suggestion
		using condition_list_t = std::vector<ConditionNode>;
```
We need to preserve the order of conditions in the defines so they show up properly in tooltips, plus there's no benefit to using a `std::set` since there aren't any duplicates we're worried about. This should also fix (some of) the Mac compile errors, which are to do with `std::set` needing to be able to do `<` less than comparisons on its elements.